contracts/apps/AragonApp.sol

```diff
@@ -3,15 +3,29 @@ pragma solidity ^0.4.18;
 import "./AppStorage.sol";
 import "../common/Initializable.sol";
 import "../evmscript/EVMScriptRunner.sol";
+import "../acl/ACLSyntaxSugar.sol";
 
 
-contract AragonApp is AppStorage, Initializable, EVMScriptRunner {
+contract AragonApp is AppStorage, Initializable, ACLSyntaxSugar, EVMScriptRunner {
     modifier auth(bytes32 _role) {
-        require(canPerform(msg.sender, _role));
+        require(canPerform(msg.sender, _role, new uint256[](0)));
         _;
     }
 
-    function canPerform(address _sender, bytes32 _role) public view returns (bool) {
-        return address(kernel) == 0 || kernel.hasPermission(_sender, address(this), _role, new bytes(0));
+    modifier authP(bytes32 _role, uint256[] params) {
+        require(canPerform(msg.sender, _role, params));
+        _;
+    }
+
+    function canPerform(address _sender, bytes32 _role, uint256[] params) public view returns (bool) {
+        bytes memory how; // no need to init memory as it is never used
+        if (params.length > 0) {
+            uint256 byteLength = params.length * 32;
+            assembly {
+                how := params // forced casting
+                mstore(how, byteLength)
+            }
+        }
+        return address(kernel) == 0 || kernel.hasPermission(_sender, address(this), _role, how);
```

Comment:
This should probably be `address(kernel) != 0 && ...`?

`getCode()` will fail if `address(kernel) == 0` so you won't be able to perform the action anyway.

Reply:
this is for the case when you are using the app without a proxy